docs/design/storagenode-graceful-exit/overview.md

```diff
@@ -0,0 +1,169 @@
+# Storage Node Graceful Exit Overview
+
+## Abstract
+
+When a storage node wants to leave the network, we would like to ensure that the pieces get transferred to other nodes. This avoids additional repair cost and bandwidth. This design document describes the necessary components for the process.
+
+## Background
+
+Storage nodes may want to leave the network. Taking a storage node directly offline would mean that the satellite needs to repair the lost pieces. To avoid this we need a process to transfer the pieces to other nodes before taking the exiting node offline. We call this process Graceful Exit. Since this process can take time we need an overview of the nodes exiting and the progress.
+
+A storage node may also want to exit a single satellite rather than the whole network. For example, the storage node doesn't have as much bandwidth available or a satellite's payment structure has changed.
+
+The process must also consider that the storage nodes may have limited bandwidth available or time to exit. This means the process should prioritize pieces with lower durability.
+
+A storage node will no longer participate in repairs or uploads once a Graceful Exit has been initiated. In addition, bandwidth used for piece transfers should not count against bandwidth used.
+
+The exiting node must continue to respond to audit and download requests.
```

Comment:
And retain requests - we do not want it to keep pieces that might have already been deleted from the satellite it is exiting from.

Reply:
Updated